src/utils/internalLinks.js

```diff
@@ -120,7 +120,7 @@ export const decodeHashComponent = (string: string): string => {
 /** Parse the operand of a `stream` operator, returning a stream name. */
 const parseStreamOperand = (operand, streamsById): string => {
   // "New" (2018) format: ${stream_id}-${stream_name} .
-  const match = /^(\d+)-/.exec(operand);
+  const match = /^(\d+)(-.*)?/.exec(operand);
   if (match) {
     const stream = streamsById.get(parseInt(match[0], 10));
```

Comment:
Hmm, in fact! I think something is slightly wrong with the code before the change made in this PR. In particular, `match[0]`, here, will be what gets matched by the _entire pattern_, which means it might include `-`. Or, after the change in this commit, it might include, e.g., `-asdfjkl;` (if "asdfjkl;" is the stream name). Oddly, both `parseInt('123-', 10)` and `parseInt('123-asdfjkl;', 10)` seem to give `123`—but that seems like too odd of a quirk to rely on.

What `parseInt` wants (or should want, anyway), instead of `match[0]`, is just the stream ID string, with none of the stuff after it. Handily, the parentheses around the just-stream-ID-string pattern—i.e., the `(` and `)` in `(\d+)`—make a "capturing group", which I've mentioned in a comment above.

Capturing groups will be stored in `match` in order of their appearance in the pattern, starting from index 1 (note: not index 0, which is reserved for what the entire pattern matched). 

I find https://regex101.com/ quite helpful in debugging regular expressions, just make sure to select "EcmaScript (JavaScript)" instead of the default "PCRE (PHP)". 🙂

So, to fix (unless @gnprice has a different idea): Just a simple commit, [separate](https://zulip.readthedocs.io/en/latest/git/fixing-commits.html) from the one you've already made, with the following change:

```diff
- const stream = streamsById.get(parseInt(match[0], 10));
+ const stream = streamsById.get(parseInt(match[1], 10));
```

I see that the web app already has [`m[1]`](https://github.com/zulip/zulip/blob/08d716c/static/js/stream_data.js#L320), instead of `m[0]`, in its corresponding code.

Reply:
Yes, I noticed it earlier too but it didn't change it because it directly impact this issue (or so I thought). I have made that change. I also changed the test cases so that it passes now. Thanks for the help!